pkg/admission/validate_bucketclass.go

```diff
@@ -0,0 +1,92 @@
+package admission
+
+import (
+	"encoding/json"
+
+	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	"github.com/noobaa/noobaa-operator/v5/pkg/bucketclass"
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+	"github.com/sirupsen/logrus"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// BucketClassValidator is the context of a bucketclass validation
+type BucketClassValidator struct {
+	Logger     *logrus.Entry
+	arRequest  *admissionv1.AdmissionReview
+	arResponse *admissionv1.AdmissionReview
+}
+
+// NewBucketClassValidator initializes a BucketClassValidator to be used for loading and validating a bucketclass
+func NewBucketClassValidator(arRequest admissionv1.AdmissionReview) *BucketClassValidator {
+	bcv := &BucketClassValidator{
+		Logger:    logrus.WithField("admission bucketclass validation", arRequest.Request.Namespace),
+		arRequest: &arRequest,
+		arResponse: &admissionv1.AdmissionReview{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "AdmissionReview",
+				APIVersion: "admission.k8s.io/v1",
+			},
+			Response: &admissionv1.AdmissionResponse{
+				UID:     arRequest.Request.UID,
+				Allowed: true,
+				Result: &metav1.Status{
+					Message: "allowed",
+				},
+			},
+		},
+	}
+	return bcv
+}
+
+// ValidateBucketClass call appropriate validations based on the operation
+func (bcv *BucketClassValidator) ValidateBucketClass() admissionv1.AdmissionReview {
+	switch bcv.arRequest.Request.Operation {
+	case admissionv1.Create:
+		bcv.ValidateCreate()
+	default:
+		bcv.Logger.Error("Failed to identify bucketclass operation type")
```

Comment:
this will be printed for any operation other than Create. can make the logs confusing

Reply:
In the ValidatingWebhookConfiguration file, only CREATE operations for bucketclass resource should be webhooked, so any other operation shouldn't even be processed by the admission server, the default case here is just best practice design to be safe.

Reply:
same goes for the resource identification switch case in `validator.go`